### PR TITLE
FIX: show error msg on grant badge if message available from backend

### PR DESCRIPTION
**Pull request by mrfinch**
https://meta.discourse.org/t/cant-grant-badge-generic-error/66982/9

### CODE REVIEW

File: app/assets/javascripts/admin/controllers/admin-user-badges.js.es6 [hunk 1]
```diff
@@ -70,9 +70,20 @@ export default Ember.Controller.extend(GrantBadgeController, {
             }
           });
         },
-        function() {
-          // Failure
-          bootbox.alert(I18n.t("generic_error"));
+        function(error) {
```
Comment (eviltrout) [inline app/assets/javascripts/admin/controllers/admin-user-badges.js.es6]: wouldn't our `popupAjaxError` helper do all this logic?

Comment (mrfinch) [reply app/assets/javascripts/admin/controllers/admin-user-badges.js.es6]: I didn't know about this function. That function works here perfectly though had to fix a bug in it :slightly_smiling_face:

File: app/assets/javascripts/discourse/lib/ajax-error.js.es6 [hunk 2]
```diff
@@ -32,7 +32,7 @@ export function extractError(error, defaultMessage) {
     } else if (parsedJSON.error) {
       parsedError = parsedJSON.error;
     } else if (parsedJSON.failed) {
-      parsedError = parsedJSON.message;
+      parsedError = parsedJSON.failed;
```
Comment (SamSaffron) [inline app/assets/javascripts/discourse/lib/ajax-error.js.es6]: careful ... this is changing internal behaviour, I guess we kind of mucked up not having this documented in a test. 

https://github.com/discourse/discourse/blob/7b253dbe4a92f7656bbc779d68e07ec93c56dfed/app/controllers/admin/backups_controller.rb#L40-L40

Comment (mrfinch) [reply app/assets/javascripts/discourse/lib/ajax-error.js.es6]: I have fixed it now.

**Pull request merged**